File: app/src/test/java/basic/option/_utils/UtilsTest.java
package basic.option._utils;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

import android.content.res.Resources;

import org.junit.jupiter.api.Test;

import org.breezyweather.R;
import org.breezyweather.common.basic.models.options._basic.Utils;

public class UtilsTest {

    @Test
    public void getNameByValue() {
        Resources res = mock(Resources.class);
        when(res.getStringArray(R.array.dark_modes)).thenReturn(new String[] {
                "Automatic", "Follow system", "Always light", "Always dark"
        });
        when(res.getStringArray(R.array.dark_mode_values)).thenReturn(new String[] {
                "auto", "system", "light", "dark"
        });
        assertEquals(
                Utils.INSTANCE.getNameByValue(res, "auto", R.array.dark_modes, R.array.dark_mode_values),
                "Automatic"
        );
    }
}

File: app/src/test/java/org/breezyweather/option/_utils/UtilsTest.kt
package org.breezyweather.option._utils

import android.content.res.Resources
import io.kotest.matchers.shouldBe
import io.mockk.every
import io.mockk.mockk
import kotlinx.coroutines.test.runTest
import org.breezyweather.R
import org.breezyweather.common.basic.models.options._basic.Utils
import org.junit.jupiter.api.Test

class UtilsTest {
    @Test
    fun getNameByValue() = runTest {
        val res = mockk<Resources>()
        every { res.getStringArray(R.array.dark_modes) } returns arrayOf("Automatic", "Follow system", "Always light", "Always dark")
        every { res.getStringArray(R.array.dark_mode_values) } returns arrayOf("auto", "system", "light", "dark")
        Utils.getNameByValue(res, "auto", R.array.dark_modes, R.array.dark_mode_values) shouldBe "Automatic"
    }
}
